src/EventSubscriber/LinkTypePlusEventSubscriber.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\commerce_gmo_linktypeplus\EventSubscriber;
+
+use Drupal\commerce_gmo_linktypeplus\Event\LinkTypePlusEvent;
+use Drupal\commerce_order\Entity\Order;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Orderstatus change event subscriber.
+ */
+class LinkTypePlusEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
+   *   Logger .
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, LoggerChannelFactoryInterface $loggerFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->loggerFactory = $loggerFactory->get('linktype_event_subscriber');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'Credit' => 'onlinkTypePaymentEvent',
```

Comment:
We need an improvement here. While gmo sending the data , the status response varies based on the payment method.

For eg: If credit card, the status will be

AUTH: Provisional sales
CAPTURE: Immediate sales
UNPROSESSED: Unsettled (payment has not been completed due to authorization error, etc.)
AUTHENTICATED: Unsettled (3D registered)

If it is paypay , the status will be 
AUTH: Authorization
SALES: Sales capture
CAPTURE: Instant capture
CANCEL: Cancel
RETURN: Refund
PAYFAIL: Payment failure (payment failed due to an error)

We have to map this gmo status with our drupal payment workflow state.  Similar to this we have many payment methods , as of now we wont be able to map all of them . So we can implement status Mapping for paypay / cvs / paypal now.

Note: Others should be able to extend this based on their payment method.

We can discuss this